lucene/core/src/java/org/apache/lucene/index/EagerCheapMergePolicy.java

```diff
@@ -0,0 +1,202 @@
+                "Found a cheap merge: " + toSummaryString(cheapMerge.segments));
+            //FYI further IW infoStream message "registerMerge" will have segment details
+          }
+          MergeSpecification spec = new MergeSpecification();
+          spec.add(cheapMerge);
+          return spec;
+        }
+      } else {
+        if (verbose) {
+          infoStream.message(INFO_STREAM_CATEGORY,
+              "Don't look for a cheap merge because " +
+                  mergeContext.getMergingSegments().size() + " segments are already merging.");
+        }
+      }
+    }
+    return super.findMerges(mergeTrigger, segmentInfos, mergeContext); // note: could still be null
+  }
+
+  private String toSummaryString(List<SegmentCommitInfo> segments) {
+    long sizeBytes = segments.stream().mapToLong(EagerCheapMergePolicy::size).sum();
+    return String.format(Locale.ROOT, "segCount=%d sizeMB=%.3f", segments.size(), sizeBytes / 1024.0 / 1024.0);
+  }
+
+  private OneMerge findCheapMerge(SegmentInfos infos, MergeContext mergeContext) throws IOException {
+    List<SegmentCommitInfo> cheapMerge = findCheapMerge(infos.asList(), mergeContext.getMergingSegments());
+    return cheapMerge == null ? null : new OneMerge(cheapMerge);
+  }
+
+  //@VisibleForTesting
+  List<SegmentCommitInfo> findCheapMerge(List<SegmentCommitInfo> segmentInfoPerCommits,
+                                         Collection<SegmentCommitInfo> alreadyMerging) {
+
+    // Find segments that are below the size threshold and that aren't being merged already
+    final List<SegmentCommitInfo> infosSortedAsc = segmentInfoPerCommits.stream()
+        .filter(o -> !alreadyMerging.contains(o))
+        .filter(o -> size(o) <= cheapMergeThresholdBytes)
+        .sorted(Comparator.comparingLong(EagerCheapMergePolicy::size))
+        .limit(getMaxMergeAtOnce())
+        .collect(Collectors.toList());
+
+    // Maximize how many of these we can merge while keeping in budget
+    long accumSize = 0L;
+    int numSegmentsToMerge = 0;
+    for (int segIdx = 0; segIdx < infosSortedAsc.size(); segIdx++) {
+      SegmentCommitInfo infoPerCommit = infosSortedAsc.get(segIdx);
+      long segSize = size(infoPerCommit);
+      if (accumSize + segSize > cheapMergeThresholdBytes) {
+        break;
+      }
+      // if the ratio of this segment's size to the sum of sizes of segments we've seen is less than
+      //   a threshold, then decide we will merge this (along with all before).
+      // note: on 1st loop, accumSize==0; division results in: Infinity < whatever (false)
+      if (((double) (segSize)) / ((double) accumSize) < cheapMaxSizeRatio) {
+        numSegmentsToMerge = segIdx + 1;
+      }
+      accumSize += segSize;
+    }
+    if (numSegmentsToMerge < cheapMinMergeAtOnce) {
+      return null; // too few segments to merge
+    }
+    return infosSortedAsc.subList(0, numSegmentsToMerge);
+  }
+
+  private static long size(SegmentCommitInfo infoPerCommit) {
```

Comment:
Rename to `sizeInBytes`?

Reply:
Intentionally this is named consistently with `MergePolicy.size()`.  Otherwise I'd prefer the unit as well.

Reply:
Let's not propagate the bad naming?  Just because an existing method is poorly named doesn't mean we should name new methods poorly too :)  Maybe open an issue to deprecate/rename `MergePolicy.size`?

Also, why not just use `MergePolicy.size` instead of adding a new dangerous method?